### ISSUE TITLE
fromAnything - community project idea

### ISSUE THREAD

**Issue by Swizz** [cujojs/most#340]
Hi everybody !

_I really do not know if I am in the right way by writing this here._

Fallen in love about reactive programing, then met MostJs made my hear beating a lot.
So I wished a way to perform something I did with Xstream : build a stream for a Firebase producer.

However, most seem to do not have this feature (unless I badly sought).
That is why I found another way. I used proxy and event capabilities : 
```js
most.fromFirebase = function(event_name, reference_obj) {
  const event_emitter = new EventEmitter();
  
  reference_obj.on(event_name, 
    (snapshot) =>  event_emitter.emitEvent(event_name, [snapshot])
  )
	
  return most.fromEvent(event_name, event_emitter);
}
```

I know this isnt the better way. I read in code about `StreamSource`.

So, this is not the main subject. This research work, raised something in my mind. If I want tomorrow to move to GunJs or the departed Parse, I will need to do the same research and build adapter again and again.
And what about socketIO ? And others ?

**_TL;DR is here_**
Therefore what are you thinking about building a massive libraries of forAnything stream builder ?
Somewhere to place all our work for building stream for any source.

I guess, one bit dependency for just a source, could be a lot, so what about building many packages but in the same repository like Cyclejs.
Can I use `@most` npm scope ? With something like `@most/from-sourcename`

Cheers,

**Comment by TylorS** [comment]
Hello @Swizz, welcome to mostjs I hope you find our libraries to be useful to you.

If I'm not mistaken you're looking for something like [`@most/create`](https://github.com/mostjs/create) which is analogous to `xs.create(producer)` in xstream. I've use firebase in a similar fashion to this

```js
import create from '@most/create'

function fromFirebase (eventName, firebaseReference) {
  return create((add, end, error) => {
    firebaseReference.on(eventName, add)
    return () => firebaseReference.off(eventName, add)
  })
}
```

>  what about building many packages but in the same repository like Cyclejs

We've discussed the potential of working in a monorepo, but there are some drawbacks I've experienced with Cycle.js as well (I'm a core contributor there too), namely extremely long build times on TravisCI. The Cycle.js tests take about 18-20 minutes on Travis CI which is not fun for outside contributors, because it can take a long time to find out if your test failed, and then make a trivial change and again have to wait a large amount of time to see if it has passed with your new change. 

Most of the time is spent on installing dependencies in each of the packages, and then in the case of Cycle.js we run `@cycle/dom` against a multitude of browsers with saucelabs to ensure cross-browser support. Some of these things could potentially be non-problems for most, but we've not made this decision yet.

> Can I use @most npm scope

The `@most` scope is for packages found [in the mostjs orginization](https://github.com/mostjs) which is the home for packages that the mostjs core team are committed to maintaining ourselves. So if we find a package useful and we'd like to continue maintenance ourselves in the event the author is not active then yes. In general community contributions are suggested to be published with a `most-` prefix and we generously welcome contributions to find a home in our [community organization](https://github.com/mostjs-community). I myself maintain `most-subject` and `most-proxy` there.

**Comment by Swizz** [comment]
Hello @TylorS,

I apologize to have not made so much research.  It is what I looking for.

Thank you very much for all the answers.
I will read more about github/most and github/mostjs-community works.

And I apologize again.

**Comment by TylorS** [comment]
No need for apologies, glad to help you find it, and I hope you enjoy using *most.js* :slightly_smiling_face:
